### DISCUSSION TITLE
How to run the generate command?

### DISCUSSION THREAD

**Discussion by RobertWalterU3d** [eclipse-langium/langium#631] [Q&A] [answered]
Here's a beginner's question: I'm looking at the Getting started guide and want to try out the "generate" command that is defined as cli tool, IIUC. How do I actually issue that command? 
I've tried running it from the VS Code terminal, but it is not known. It is also not visible when searching for it in "Show All Commands".   What do I need to do to run the language's generate command?

**Comment by msujew** [comment, marked as answer]
Hey @RobertWalterU3d,

The yeoman generator should generate a `bin/cli` file in your project folder. You can simply run it using `node bin/cli`. The project is configured so that if you for example run `npm link` (which globally links your project, so that it can be consumed by other npm packages), you can just run `<packagename> generate` in another npm project. This will automatically run the `bin/cli` file from your DSL project.

**Reply by RobertWalterU3d** [reply]
Okay, I also found this bullet point in the file "langium-quickstart.md", but it doesn't work for me:

>  * Run `./bin/cli` to see options for the CLI; `./bin/cli generate <file>` generates code for a given DSL file.

When I put that in the terminal, like so:

> ./bin/cli generate ./sandbox/Demo.mydsl

, it prompts me to select a program with which I want to open the file .bin/cli 

This, however, works now:

> node bin/cli generate sandbox/Demo.mydsl

I've never used npm or node before really, so this is all a bit arcane to me.

**Reply by msujew** [reply]
I see, yeah we should change this. You're probably running Windows, right? On Mac/Linux systems this works, but Windows doesn't like that.

**Reply by RobertWalterU3d** [reply]
That is correct, I'm on Windows

**Reply by RobertWalterU3d** [reply]
I can't see how I can mark your initial reply as "Answer". But it should marked as such :) Thanks again!

**Reply by msujew** [reply]
The category of the discussion was `General` which doesn't support the answering mechanic. I've changed this into a `Q&A`. Anyway, you're welcome :)
